### PR TITLE
Customized ALB actions

### PR DESCRIPTION
**Pull request by bigkraig**
Addresses #487. Provide a method for defining custom ALB rule actions via annotations.

### CODE REVIEW

**Comment by bigkraig** [comment]
Still need to implement support on default backends, but everything else should be in order.

File: docs/ingress-resources.md [hunk 1]
```diff
@@ -110,6 +109,8 @@ Optional annotations are:
 
 - **ssl-policy**: Defines the [Security Policy](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies) that should be assigned to the ALB, allowing you to control the protocol and ciphers.
 
+- **alb.ingress.kubernetes.io/actions.\<ACTION NAME>**: Provides a method for configuring custom actions on a listener, such as for [Redirect Actions](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-listeners.html#redirect-actions). The `<ACTION NAME>` in the annotation must match the `serviceName` in the ingress rules. The value of the annotation is the JSON spec of the action. See the [Action type](https://docs.aws.amazon.com/sdk-for-go/api/service/elbv2/#Action) for documentation on what should be in the JSON. An example for a fixed-response would be: `alb.ingress.kubernetes.io/actions.fixed-response-error: '{"Type": "fixed-response", "FixedResponseConfig": {"ContentType":"text/plain", "StatusCode":"503", "MessageBody":"503 error text"}}'` for a `serviceName` of `fixed-response-error`. _NOTE_ you must set the `servicePort` to `use-annotation`.
```
Comment (anhpham1509) [inline docs/ingress-resources.md]: I think it's better to have examples and notes of `fixed-respone` and `redirect` clearly in sub-indented bullet points.

File: internal/alb/rs/rule_test.go [hunk 2]
```diff
@@ -22,8 +25,39 @@ func TestNewDesiredRule(t *testing.T) {
...
+			TargetPort: 0,
+			ExpectedRule: Rule{
+				svc: svc{desired: service{name: "fixed-response-action", port: intstr.FromString("use-annotation"), targetPort: 0}},
+				rs: rs{
+					desired: &elbv2.Rule{
+						Priority:  aws.String("default"),
+						IsDefault: aws.Bool(true),
+						Actions: []*elbv2.Action{
+							{
+								Type: aws.String("fixed-response"),
```
Comment (anhpham1509) [inline internal/alb/rs/rule_test.go]: 1 test case for `redirect` also?

File: internal/ingress/annotations/action/main.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	return &Config{
+		Actions: actions,
+	}, nil
+}
+
+func Dummy() *Config {
+	return &Config{
+		Actions: map[string]*elbv2.Action{
+			"fixed-response-action": &elbv2.Action{
+				Type: aws.String("fixed-response"),
```
Comment (anhpham1509) [inline internal/ingress/annotations/action/main.go]: How's about dummy type `redirect`? :-D

File: docs/ingress-resources.md (hunk 1 above)
Comment (gabzim) [reply docs/ingress-resources.md]: One of the main reasons we're waiting for this is http -> https redirection. Which I'd say is the biggest use case so far. Maybe that should be a redirect example?

**Pull request merged**